### ISSUE TITLE
Only allow table columns for grouping where it makes sense

### ISSUE THREAD

**Issue by wrybind** [grocy/grocy#1535]
In some cases of grouping the stock entries, the group headers get replaced by "[object Object]". It happens sometimes when grouping by a category that is also displayed in a column, but not always.

How to reproduce on the demo:
1. Go to stock overview
2. Reset table options
3. Group by "Amount"

![2021-07-18 12_22_59-Window](https://user-images.githubusercontent.com/79023505/126064119-eb65251e-9224-4eaa-8d57-b617acbdd6b9.png)

**Comment by berrnd** [comment]
It depends on the source data, I doubt that it really makes sense to group by amount (or calories, or the last price, or the product picture - or any column which mostly contains unique values).

Currently just any column is allowed for grouping, of course this can be changed to only allow the appropriate ones and them where the source data allows to do it properly...

**Comment by wrybind** [comment]
Yes, I agree. Grouping by amount does not really make sense. The behaviour occured first for me, when I was grouping by product group and playing around with it. Sorting in different directions, by different columns, etc. Suddenly the product group headers disappeared and were replaced by object Object. However, I could not reproduce that and find out how I got there. Grouping by amount was the easiest way to showing what happens on the demo.
